src/rules/parser.go

```diff
@@ -177,6 +177,16 @@ func (p *parser) parseRuleInfo(st ir.Node, labelStmt ir.Node, proto *Rule) (Rule
 			}
 			rule.Name = part.Params[0]
 
+		case "link":
+			if len(part.Params) != 1 {
+				return rule, p.errorf(st, "@link expects exactly 1 param, got %d", len(part.Params))
+			}
+			var link = part.Params[0]
+			if !isURL(link) {
```

Comment:
you've really overcomplicated the logic. Why is there a check for `url`?

Reply:
After discussion, I removed it completely